doc/design/staging-1.md

```diff
@@ -0,0 +1,37 @@
+# Staging Environment and Procedures
+
+## Production Environment
+* Django API server, Postgres, and CloudAMQP running in Heroku
+* Sponsored S3 bucket in the sponsored account in AWS
+* Web GUI served through Netlify
+* Sentry instance
+
+## Staging Environment
+* Django API server, Postgres, and CloudAMQP running in Heroku
+* Private S3 bucket in the project account in AWS
+* Web GUI served through Netlify
+* Sentry instance
+
+This should exactly parallel production, except for the S3 bucket, which is private.
+Nothing is actually shared, everything is hosted adjacently.
+
+The staging server will contain some test data and some (smallish) real data that reflects the contents of production.
+
+## Release procedure
+The staging server will track the `master` branch in `dandi-api` and will automatically redeploy whenever `master` is changed.
+
+The production server will not track any branch. Instead, it will automatically redeploy whenever a release is tagged in GitHub.
```

Comment:
Consistency sounds nice. I don't have strong feelings about the versioning scheme.

@waxlamp mentioned something about GitHub dispatching or something that would potentially make the release process cleaner?

Reply:
> GitHub does not create annotated tags for releases--why is that?

my guess so that you could adjust description without affecting the tag.  But if you push annotated tag, and just "Edit/Save" - it would produce a "proper" GitHub release based on that annotated tag. Have been doing that for many projects, never "minted" a release "from scratch" on github.